config/x_ac_platform_compat.m4

```diff
@@ -44,3 +44,17 @@ AC_DEFUN([X_AC_SLURM_MPAO_WR], [
   fi
   AC_MSG_RESULT($enable_mpao_workaround)
 ])
+
+AC_DEFUN([X_AC_SHORT_HOSTNAME], [
+  AC_MSG_CHECKING([])
```

Comment:
This looks good. A minor nit: perhaps we can also add a short description on what this check does in AC_MSG_CHECKING? (like "whether to enable short hostname match tests")?